src/snowflake/snowpark/session.py

```diff
@@ -593,6 +594,8 @@ def close(self) -> None:
         finally:
             try:
                 self._conn.close()
+                if self._temp_table_auto_cleaner.is_alive():
```

Comment:
there is one thing that i am not so clear about the behavior, let's say user created more than one session, and one session didn't get used for long time, does it gets garbage collected()? with that extra clean up component, will it block the session to be garbage collected()? do you have an answer for this?

Reply:
https://github.com/snowflakedb/snowpark-python/blob/7c9df8446301c156248d76d45b67a5d0fa7b6726/src/snowflake/snowpark/session.py#L256 will be called before Session object gets garbage collected. So with `close`, this thread will also be terminated and won't interrupt  Session object's garbage collection